docs/design-docs/modular-cli/moduler-cli.md

```diff
@@ -0,0 +1,177 @@
+    "infra apply -f data.yaml"?
+    "Infrastructure.apply(...)"
+    Task: Create/Update infrastucture on AWS/Azure/Google...
+    Input:  Yaml with at least epiphany-cluster + possible infra docs
+    Output: manifest, ansible inventory and terrafrom files
+    ```
+3. analyse:
+    ```
+    "epicli infra analyse -f data.yaml"
+    "infra analyse -f data.yaml"?
+    "Infrastructure.analyse(...)"
+    Task: Analysing existing infrastructure
+    Input:  Yaml with at least epiphany-cluster + possible infra docs
+    Output: manifest, ansible inventory
+    ```
+4. destroy:
+    ```
+    "epicli infra destroy -b /buildfolder/"
+    "infra destroy -b /buildfolder/"?
+    "Infrastructure.destroy(...)"
+    Task:  Destroy all infrastucture on AWS/Azure/Google?
+    Input:  Build folder with manifest and terrafrom files
+    Output: Deletes the build folder.
+    ```
+
+### Repository
+
+Module for creating and tearing down a repo + preparing requirements for offline installation.
+
+Functionality (rough outline and subjected to change):
+
+1. template:
+    ```
+    "epicli repo template -f outfile.yaml  (--all)"
+    "repo template -f outfile.yaml (--all)"?
+    "Repository.template(...)"
+    Task: Generate a template yaml for a repository
+    Input:  File to output data, provider and possible all flag
+    Output: outfile.yaml template
+    ```
+2. prepare:
+    ```
+    "epicli repo prepare -os (ubuntu-1904/redhat-7/centos-7)"
+    "repo prepare -o /outputdirectory/"?
+    "Repo.prepare(...)"
+    Task: Create the scripts for downloading requirements for a repo for offline installation for a certain OS.
+    Input:  Os which we want to output the scripts for:  (ubuntu-1904/redhat-7/centos-7)
+    Output: Outputs the scripts  scripts
+    ```
+3. create:
+    ```
+    "epicli repo create -b /buildfolder/ (--offline /foldertodownloadedrequirements)"
+    "repo create -b /buildfolder/"?
+    "Repo.create(...)"
+    Task: Create the repository on a machine (either by running requirement script or copying already prepared ) and sets up the other VMs/machines to point to said repo machine. (Offline and offline depending on --offline flag)
+    Input:  Build folder with manifest and ansible inventory and posible offline requirements folder for onprem installation.
+    Output: repository manifest or something only with the location of the repo?
+    ```
+4. teardown:
+    ```
+    "epicli repo teardown -b /buildfolder/"
+    "repo teardown -b /buildfolder/"?
+    "Repo.teardown(...)"
+    Task: Disable the repository and resets the other VMs/machines to their previous state.
+    Input:  Build folder with manifest and ansible inventory
+    Output: -
+    ```
+
+### Components
+
+Module for applying a command on a component which can contain one or multiple roles. It will take the Ansible inventory to determin which roles should be applied to which component. The command each role can implement are (rough outline and subjected to change):

```

Comment:
```suggestion
Module for applying a command on a component which can contain one or multiple roles. It will take the Ansible inventory to determine which roles should be applied to which component. The command each role can implement are (rough outline and subjected to change):
```